homeassistant/components/elkm1/__init__.py

```diff
@@ -144,24 +189,37 @@ def _included(ranges, set_to, values):
     return True
 
 
-def _create_elk_services(hass, elk):
+def _create_elk_services(hass, elks):
     def _speak_word_service(service):
-        elk.panel.speak_word(service.data.get('number'))
+        prefix = service.data['prefix']
+        elk = elks.get(prefix)
+        if elk is None:
+            _LOGGER.error("No elk m1 with prefix for speak_word: '%s'",
+                          prefix)
+            return
+        elk.panel.speak_word(service.data['number'])
 
     def _speak_phrase_service(service):
-        elk.panel.speak_phrase(service.data.get('number'))
+        prefix = service.data['prefix']
+        elk = elks.get(prefix)
+        if elk is None:
+            _LOGGER.error("No elk m1 with prefix for speak_phrase: '%s'",
+                          prefix)
+            return
+        elk.panel.speak_phrase(service.data['number'])
 
     hass.services.async_register(
         DOMAIN, 'speak_word', _speak_word_service, SPEAK_SERVICE_SCHEMA)
     hass.services.async_register(
         DOMAIN, 'speak_phrase', _speak_phrase_service, SPEAK_SERVICE_SCHEMA)
 
 
-def create_elk_entities(hass, elk_elements, element_type, class_, entities):
+def create_elk_entities(elk_data, elk_elements, element_type,
+                        class_, entities):
     """Create the ElkM1 devices of a particular class."""
-    elk_data = hass.data[DOMAIN]
     if elk_data['config'][element_type]['enabled']:
         elk = elk_data['elk']
+        _LOGGER.debug("create_elk_entities for %s", elk)
```

Comment:
We start logging messages with capital letter. Don't log the function name. Write out what is done.
```suggestion
        _LOGGER.debug("Creating elk entities for %s", elk)
```

Reply:
ok